lib/mastodon/snowflake.rb

```diff
@@ -138,10 +124,10 @@ def ensure_id_sequences_exist
       end
     end
 
-    def id_at(timestamp)
-      id  = timestamp.to_i * 1000 + rand(1000)
+    def id_at(table_name, timestamp)
+      id  = timestamp.to_i * 1000
       id  = id << 16
-      id += rand(2**16)
+      id |= ApplicationRecord.connection.select_value('SELECT timestamp_id($1)', nil, [[nil, table_name]]) & 65_535
```

Comment:
Does this really get us anything over the `rand(2**16)`? The returned ID is almost certainly not for the millisecond we're calling `id_at` for (as it's almost certainly being called for a historical time, not the current time), which means it's roughly random. If it's called several times in the same millisecond, we can at least be guaranteed that the low bits won't be the same, but I don't know how much that matters.